pkg/operator/ceph/controllers/nodedrain/reconcile.go

```diff
@@ -0,0 +1,64 @@
+package nodedrain
+
+import (
+	"github.com/rook/rook/pkg/operator/ceph/controllers/controllerconfig"
+
+	"github.com/coreos/pkg/capnslog"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var log = capnslog.NewPackageLogger("github.com/rook/rook", "nodedrain-controller")
+
+// ReconcileNode reconciles ReplicaSets
+type ReconcileNode struct {
+	// client can be used to retrieve objects from the APIServer.
+	scheme  *runtime.Scheme
+	client  client.Client
+	log     *capnslog.PackageLogger
+	options *controllerconfig.Options
+}
+
+// Implement reconcile.Reconciler so the controller can reconcile objects
+var _ reconcile.Reconciler = &ReconcileNode{}
+
+// Reconcile reconciles a node and ensures that it has a drain-detection deployment
+// attached to it.
+func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Info("Reconciling node")
+	// // Fetch the ReplicaSet from the cache
+	// rs := &appsv1.ReplicaSet{}
+	// err := r.client.Get(context.TODO(), request.NamespacedName, rs)
```

Comment:
When you fill in this logic, try to make sure we're getting items from the informer cache here instead of making apiserver GET requests.

Reply:
The controller-runtime client is a delegating client. It reads from cache and writes from a rest client. :)